client/src/pages/MainTable.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect } from "react";
+
+const MainTable = () => {
+  console.log("Page Uploaded");
+  const [clientData, SetClientData] = useState (null);
+  const [error, setError] = useState(null);
+  const [selectedClientId, setSelectedClientId] = useState (null);
+
+  useEffect(() => {
+    fetch("/api/clients")
+      .then((response) => {
+        if (!response.ok) {
+          throw Error("We could not fetch the data for that resource");
+          }
+          return response.json();
+        })
+      .then((data) => {
+        console.log(data,"After");
+        SetClientData(data);
+        console.log(clientData, "Data from useState");
+        setError(null);
+      })
+      .catch((err) => {
+              console.log("error message: ", err.message);
+              setError(err.message);
+      });
+  }, []);
+
+  console.log(clientData, "outside of fetch");
+
+if (clientData === null){
+  return <h1>Loading...</h1>;
```

Comment:
```suggestion
  return <p>Loading...</p>;
```
Headings are used for the titles of sections. `Loading` isn't really a title of a section, so we can just use the `p` element for now. We can replace this with a loading spinner later

Reply:
Thank you Coung, let me include the spiner in the middle.